client/src/components/Header.js

```diff
@@ -79,6 +81,10 @@ function Header() {
             type:"success"
             })   
           setAnchorEl(null)
+          userContext.setIsSubscribed(false)
```

Comment:
reset user context when logging out

Reply:
> @bugatti12345
> Now the redirect works, but it gives this error whenever i sign up/login the first time after restarting the client & server. If i try it afterwards, it's working as intended. I am not sure what the issue is here. Can you take a look at the code and see if im doing things right?
> 
> ![Screen Shot 2021-03-31 at 4 52 21 PM](https://user-images.githubusercontent.com/32600876/113210116-e22c0f80-9241-11eb-9008-a45cd59c8a0a.png)

I don't think you did anything wrong, but can you maybe try change the `redirectOnLogin()` in line 131 into `redirectOnLogin`?

Reply:
However the error is not related to redirect part tho...